include/vinecopulib/vinecop/implementation/class.ipp

```diff
@@ -146,9 +132,15 @@ inline Vinecop::Vinecop(const boost::property_tree::ptree input,
       pair_copulas_[tree][edge] = Bicop(pc_node);
     }
   }
-
-  threshold_ = 0;
-  loglik_ = NAN;
+  var_types_ = tools_serialization::ptree_to_vector<std::string>(
+    input.get_child("var_types"));
+
+  // try block for backwards compatibility
```

Comment:
Same comment as for bicop regarding having var_types in the try block (and also about the (...) {}).

Reply:
- Yeah, it should by in the try block. 
- A `try` always needs to be followed by a `catch`. Otherwise it doesn't compile.